libFMS_mod/libFMS.F90

```diff
@@ -0,0 +1,471 @@
+                       stock_move, stock_type, stock_print, get_index_range, &
+                       stock_integrate_2d
+  use stock_constants_mod, only: NELEMS, ISTOCK_WATER, ISTOCK_HEAT, ISTOCK_SALT, &
+                       ISTOCK_TOP, ISTOCK_BOTTOM, ISTOCK_SIDE, stocks_file, &
+                       stocks_report, stocks_report_init, stocks_set_init_time
+
+  !> field manager
+  use field_manager_mod, only: field_manager_init, field_manager_end, find_field_index, &
+                         find_field_index_old, find_field_index_new, get_field_info, &
+                         get_field_method, get_field_methods, parse, fm_change_list, &
+                         fm_change_root, fm_dump_list, fm_exists, fm_get_index, &
+                         fm_get_current_list, fm_get_length, fm_get_type, fm_get_value, &
+                         fm_get_value_integer, fm_get_value_logical, fm_get_value_real, &
+                         fm_get_value_string, fm_intersection, fm_init_loop, &
+                         fm_loop_over_list, fm_new_list, fm_new_value, &
+                         fm_new_value_integer, fm_new_value_logical,fm_new_value_real, &
+                         fm_new_value_string, fm_reset_loop, fm_return_root, &
+                         fm_modify_name, fm_query_method, fm_find_methods, fm_copy_list, &
+                         fm_set_verbosity, fm_field_name_len, fm_path_name_len, &
+                         fm_string_len, fm_type_name_len, NUM_MODELS, NO_FIELD, &
+                         MODEL_ATMOS, MODEL_OCEAN, MODEL_LAND, MODEL_ICE, MODEL_COUPLER, &
+                         fm_array_list_def, method_type, method_type_short, &
+                         method_type_very_short, fm_list_iter_type, default_method
+  use fm_util_mod, only: fm_util_start_namelist, fm_util_end_namelist, &
+                         fm_util_check_for_bad_fields, fm_util_set_caller, &
+                         fm_util_reset_caller, fm_util_set_no_overwrite, &
+                         fm_util_reset_no_overwrite, fm_util_set_good_name_list, &
+                         fm_util_reset_good_name_list, fm_util_get_length, &
+                         fm_util_get_integer, fm_util_get_logical, fm_util_get_real, &
+                         fm_util_get_string, fm_util_get_integer_array, &
+                         fm_util_get_logical_array, fm_util_get_real_array, &
+                         fm_util_get_string_array, fm_util_set_value, &
+                         fm_util_set_value_integer_array, fm_util_set_value_logical_array, &
+                         fm_util_set_value_real_array, fm_util_set_value_string_array, &
+                         fm_util_set_value_integer, fm_util_set_value_logical, &
+                         fm_util_set_value_real, fm_util_set_value_string, &
+                         fm_util_get_index_list, fm_util_get_index_string, &
+                         fm_util_default_caller
```

Comment:
Need to understand if these are used externally or have a valid external use before exposing them to the public via this interface.